docs/es/week12/12-2.md

```diff
@@ -0,0 +1,323 @@
+
+$$
+p (\texttt {unicornio} \mid \texttt {¿Estos ??? de pelo plateado eran previamente desconocidos})
+$$
+
+$$
+p (x_n \mid x _ {- n}) = \text {softmax} (\text {E} f (x _ {- n})))
+$$
+
+Las incrustaciones de palabras tienen cierta estructura
+
+<centro>
+<img src = "{{site.baseurl}}/images/week12/12-2/embeddings-structure.png" width = "60%" /> <br>
+<b> Fig. 9 </b>: ejemplo de estructura de inserción
+</center>
+
+- La idea es que si tomamos la incrustación de "rey" después del entrenamiento y agregamos la incrustación de "mujer" obtendremos una incrustación muy cercana a la de "reina"
+- Muestra algunas diferencias significativas entre vectores.
+
+
+#### Pregunta: ¿La representación de la palabra es dependiente o independiente del contexto?
+
+Independiente y no tengo idea de cómo se relacionan con otras palabras.
+
+
+#### Pregunta: ¿Cuál sería un ejemplo de una situación en la que este modelo tendría dificultades?
+
+La interpretación de palabras depende en gran medida del contexto. Entonces, en el caso de palabras ambiguas, palabras que pueden tener múltiples significados, el modelo tendrá dificultades, ya que los vectores de inserción no capturarán el contexto necesario para comprender correctamente la palabra.
+
+
+### GPT
+
+Para agregar contexto, podemos entrenar un modelo de lenguaje condicional. Luego, dado este modelo de lenguaje, que predice una palabra en cada paso de tiempo, reemplace cada salida del modelo con alguna otra característica.
+
+- Entrenamiento previo: predice la siguiente palabra
+- Ajuste fino: cambie a una tarea específica. Ejemplos:
+  - Predecir si es sustantivo o adjetivo
+  - Dado un texto que comprende una revisión de Amazon, predice la puntuación de sentimiento para la revisión
+
+Este enfoque es bueno porque podemos reutilizar el modelo. Entrenamos previamente un modelo grande y podemos ajustarnos a otras tareas.
+
+### ELMo
+
+GPT solo considera el contexto hacia la izquierda, lo que significa que el modelo no puede depender de ninguna palabra futura; esto limita mucho lo que el modelo puede hacer.
+
+Aquí el enfoque es entrenar _dos_ modelos de lenguaje
+
+- Uno en el texto de izquierda a derecha
+- Uno en el texto de derecha a izquierda
+- Concatenar la salida de los dos modelos para obtener la representación de la palabra. Ahora puede condicionar tanto el contexto hacia la derecha como hacia la izquierda.
+
+Esta sigue siendo una combinación "superficial" y queremos una interacción más compleja entre el contexto izquierdo y derecho.
+
+
+### BERT
+
+BERT es similar a word2vec en el sentido de que también tenemos una tarea de rellenar un espacio en blanco. Sin embargo, en word2vec teníamos proyecciones lineales, mientras que en BERT hay un gran transformador que es capaz de mirar más contexto. Para entrenar, enmascaramos el 15% de las fichas e intentamos predecir el espacio en blanco.
+
+Puede escalar BERT (RoBERTa):
+
+- Simplificar el objetivo de preentrenamiento de BERT
+- Escalar el tamaño del lote
```

Comment:
```suggestion
- Escalando el tamaño del lote
```

Reply:
Agree. Also the upper portion should change (from 'Simplificar el objetivo...' to 'SimplifiCANDO el objetivo...').